crates/iceberg/src/arrow.rs

```diff
@@ -106,3 +114,224 @@ impl ArrowReader {
         ProjectionMask::all()
     }
 }
+
+/// The key of column id in the metadata of arrow field.
+pub const COLUMN_ID_META_KEY: &str = "column_id";
```

Comment:
This is incorrect, see https://github.com/apache/iceberg-python/blob/afdfa351119090f09d38ef72857d6303e691f5ad/pyiceberg/io/pyarrow.py#L168

Reply:
This is a tricky one. In Arrow the `PARQUET:` prefix indicates that it is something specific to Parquet, in this case the [`field_id`](https://github.com/apache/parquet-format/blob/e517ac4dbe08d518eb5c2e58576d4c711973db94/src/main/thrift/parquet.thrift#L459).

Would be awesome if we could add a test for this.